### DISCUSSION TITLE
RTSP stability questions capturing the stream with FFMPEG

### DISCUSSION THREAD

**Discussion by ericmagaha** [gtxaspec/wz_mini_hacks#617] [Q&A] [answered]
Hello,

I am running wz mini hacks on 15 V3 cameras with SSH enabled, RTSP enabled, and NFS writing the video clips to my NAS.  Everything works fine for the most part.

Recently bought an IP camera (Reolink) to mess around with.  Got it working with capturing the RTSP stream using FFMPEG on my NAS and writing it to minute long files, basically to mimic the way my Wyze V3 cams are setup (even though they are using NFS).

So I figured, why not try this with a V3 cam i.e. capture the RTSP stream and write it to minute long files.

Tried it with 2 different RTSP setups on a spare V3 i.e. using the Wyze RTSP firmware, and then using 4.36.1.139 with wz mini hacks and the v4l2rtspserver RTSP setup.

It works i.e. I can capture the RTSP stream from the Wyze V3 and write it to MP4 files.  However, after some time (15 minutes, whatever) the FFMPEG process on my NAS appears to stop processing the stream, even though the FFMPEG process does not die.

I do not have this issue with the Reolink.  That FFMPEG process on my NAS has been chugging along for days and creating minute long MP4 files.

Is there something different about the Wyze V3?  Is it the iCamera process starting and stopping the wireless ethernet adapter?

My next step is to use a wired ethernet dongle on the V3.  I have had success with the UCTRONICS non-POE adapters on a few of my V3s and just ordered a UCTRONICS POE adapter.

I am thinking it is something related to WIFI.  Anyone have any experience with something like this?

TIA,

**Comment by gtxaspec** [comment]
The problem is with the camera stream and v4l2rtspserver.  Can you try the newer version with go2rtc?

**Reply by ericmagaha** [reply]
Any idea why I got similar behavior with the Wyze beta RTSP firmware using their RTSP server?  Idk what RTSP server is embedded with that firmware.  That is why I was thinking it could be related to the Wifi dropping.  I have read in other discussions here that the V3 tries to reach out to Wyze servers and/or Google and if it does not get a response it will restart the Wifi.  However, my cameras DO have access to the internet.

I will try the updated wz mini hacks and go2rtc.

**Reply by ericmagaha** [reply]
Oh dang!

Set up a spare Wyze V3 with firmware 4.36.9.139 and the latest wz mini hacks and using WIFI.  Enabled the following in wz_mini.conf:

`GO2RTC_SERVER_ENABLED="true"`
`GO2RTC_HI_RES_ENABLED="true"`
`GO2RTC_HI_RES_ENABLE_AUDIO="true"`
`GO2RTC_LOW_RES_ENABLED="true"`
`GO2RTC_LOW_RES_ENABLE_AUDIO="true"`

Can access the go2rtc web page at http://<camera_ip>:1984

Copied the high resolution 1080p link for MP4 and used that in my script.  My FFMPEG process has been chugging along now for 2 hours with no issues and writing MP4 files in 1 minute increments.

The new RTSP server i.e. go2rtc, is way more stable than the Wyze RTSP firmware 4.61.0.1 (4.61.0.3?) and way way more stable than v4l2rtspserver.

Thanks for the suggestion!

**Comment by ericmagaha** [comment, marked as answer]
> The problem is with the camera stream and v4l2rtspserver. Can you try the newer version with go2rtc?

**Comment by raelbsd** [comment]
Sorry for bringing back this issue, but I'm facing similar issues. I thought it was a ffmpeg decoding problem (you can see the ffmpeg errors in the logs) but when it looses connection it gives the same errors, so I suspect the wifi connection that can be the problem. 
I have tried wz_mini_hacks, wyze bridge and the RTSP beta firmware (4.61.0.1) and I had the best results with wz_mini_hacks (using RTSP disabled, go2rtc enabled, stand alone mode) but it's not 100% reliable. I have 8 wyze cams, with HA and Frigate standalone on a Docker server. In some cases my cams are close to my routers, that's why I discarded wifi connection's problems but maybe I'm wrong.
I'm noticing some people use a wired ethernet dongle, like this one  [https://www.amazon.com/dp/B09Z372JM4?psc=1&ref=ppx_yo2ov_dt_b_product_details](url), perhaps with this I can fix my setup?
Thanks!
